platform/ios/MapLibre.docc/LocationPrivacyExample.md

```diff
@@ -0,0 +1,132 @@
+# User Location & Location Privacy
+
+Requesting precise location with ``MLNLocationManager``.
+
+## Overview
+
+This example shows how to request a precise location with ``MLNLocationManager``.
+
+First of all you need to prepare your `Info.plist`. You need to provide a description why your app needs to access location:
+
+```plist
+<key>NSLocationWhenInUseUsageDescription</key>
+<string>Dummy Location When In Use Description</string>
+```
+
+As well as a description why your app needs precise location.
+
+```plist
+<key>NSLocationTemporaryUsageDescriptionDictionary</key>
+<dict>
+  <key>MLNAccuracyAuthorizationDescription</key>
+  <string>Dummy Precise Location Description</string>
+</dict>
+```
+
+The `Coordinator` defined below is also the ``MLNMapViewDelegate``. When the location manager authorization changes it will call the relevant method. If precise location has not been granted, a button is shown at the bottom of the map.
+
+![](ImpreciseLocation.png)
+
+When the button is pressed a pop-up with the description we set in `Info.plist` will be shown:
+
+![](PreciseLocationRequestPopup.png)
+
+<!-- include-example(LocationPrivacyExample) -->
+
+```swift
+enum LocationAccuracyState {
```

Comment:
Consider moving this example to the relevant class definition, this way you'd get it checked as well, and IDEs might be able to show it, hopefully. 
Although I'm not familiar enough with xcode, I hope some of the comments here are shown as oart of a tooltip when hovering over a class or a method.

Reply:
This is compiled. The script copies the code section into the Markdown file.